app/src/main/kotlin/at/bitfire/davdroid/push/PushNotificationManager.kt

```diff
@@ -0,0 +1,49 @@
+package at.bitfire.davdroid.push
+
+import android.accounts.Account
+import android.content.Context
+import androidx.core.app.NotificationCompat
+import at.bitfire.davdroid.R
+import at.bitfire.davdroid.ui.NotificationRegistry
+import java.util.logging.Logger
+
+object PushNotificationManager {
+    private val logger = Logger.getLogger("PushNotificationManager")
+
+    /**
+     * Generates the notification ID for a push notification.
+     */
+    private fun notificationId(account: Account, authority: String): Int {
+        return account.name.hashCode() + account.type.hashCode() + authority.hashCode()
+    }
+
+    /**
+     * Sends a notification to inform the user that a push notification has been received, the
+     * sync has been scheduled, but it still has not run.
+     */
+    fun notifyScheduled(context: Context, account: Account, authority: String) {
+        val notificationRegistry = NotificationRegistry(context, logger)
+
+        notificationRegistry.notifyIfPossible(notificationId(account, authority)) {
+            NotificationCompat.Builder(context, notificationRegistry.CHANNEL_STATUS)
+                .setSmallIcon(R.drawable.ic_sync)
+                .setContentTitle(context.getString(R.string.sync_notification_pending_push_title))
+                .setContentText(context.getString(R.string.sync_notification_pending_push_message))
+                .setPriority(NotificationCompat.PRIORITY_LOW)
+                .setCategory(NotificationCompat.CATEGORY_STATUS)
+                .setAutoCancel(true)
+                .setOngoing(true)
```

Comment:
There's no actual status nor is it important for us that the notification stays shown, so I'd remove the "ongoing" flag.

However I'd set an Intent that jumps into the DAVx5 account overview (or the specific account, don't know what's better). Then if some taps on the notification, DAVx⁵ comes into the foreground and the sync should be done (almost) immediately.

Hm we could set the account name as sub-text like in `SyncManager`:
https://github.com/bitfireAT/davx5-ose/blob/a1148613e9f31d0adf8915062d57d449868f9001/app/src/main/kotlin/at/bitfire/davdroid/sync/SyncManager.kt#L838-L840
and the Intent to open the respective account screen.